objectstore-service/src/metadata.rs

```diff
@@ -0,0 +1,46 @@
+use serde::Deserialize;
+use std::time::Duration;
+
+/// Denotes the storage usecase
+///
+/// Different decisions can be made based on this usecase, such as using a different underlying
+/// storage backend.
+#[derive(Debug, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "snake_case")]
+pub enum Usecase {
+    Attachments,
+}
+
+/// The storage scope for each object
+///
+/// Each object is stored within a scope. The scope is used for access control, as well as the ability
+/// to quickly run queries on all the objects associated with a scope.
+/// The scope could also be used as a sharding/routing key in the future.
+///
+/// While the storage service offers organization and project scopes, it does not maintain any
+/// association between those two, and it is up to the user to provide all projects belonging to an
+/// organization.
+#[derive(Debug, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "snake_case")]
+pub enum Scope {
+    Organization(u64),
+    Project(u64),
+}
```

Comment:
+1 on making this an enum so we can extend this in the future. A few questions around this:

1. Should the project scope also include the organization? For example, if I save an object for a project and then query it with organization-wide authorization, I should also be able to read it. For this, the object needs to know which organization it belongs to. Potentially, this needs to become a struct with optional fields, instead?
2. In addition to the scope with ID, should we also expose an enumeration for the kind of scope ("organization" vs "project")?
3. This serializes using the default external tagging as `{"organization": 4711}`. Would we consider custom serialization, such as `"org:4711"` to use this easily in string-only places like request headers?